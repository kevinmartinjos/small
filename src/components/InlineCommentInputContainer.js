import React from 'react';
import {Button} from 'react-bootstrap';
import FakeServer from '../FakeServer/FakeServer';

var PropTypes = React.PropTypes;

var InlineCommentInputContainer = React.createClass({
	propTypes: {
		submitHandlerCallback: PropTypes.func.isRequired,
		cancelCallback: PropTypes.func.isRequired
	},
	contextTypes: {
		postId: PropTypes.string,
		selectedText: PropTypes.string
	},
	getInitialState(){
		return {
			value: ''
		};
	},
	getDefaultState(){
		return {
			value: ''
		};
	},
	submitHandler(event){
		var server = new FakeServer();
		server.addPostComment(this.context.postId, this.state.value, this.context.selectedText);
		this.props.submitHandlerCallback();
		this.setState(this.getDefaultState());
		event.preventDefault();
	},
	handleChange(event){
		console.log(this.context.selectedText);
		this.setState({value: event.target.value});
	},
	handleCancel(){
		this.props.cancelCallback();
	},
	render() {
		return(
			<div className="InlineCommentInputContainer">
				<div>
					{this.context.selectedText}
				</div>
				<form onSubmit={this.submitHandler}>
					<div>
						<textarea onChange={this.handleChange}/>
					</div>
					<Button bsSize='sm' type='submit'>Submit</Button>
					<Button bsSize='sm' onClick={this.handleCancel}>cancel</Button>
				</form>
			</div>
		);
	}
});

export default InlineCommentInputContainer;
