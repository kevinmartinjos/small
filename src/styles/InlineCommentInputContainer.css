.InlineCommentInputContainer{
	min-width: 300px;
}

.InlineCommentInputContainer .annotation{
	font-weight: 400;
	font-size: 16px;
	font-style: normal;
	color: rgb(0, 0, 0, 0.8);
	letter-spacing: 0.01rem;
	line-height: 1.4;
	font-family: 'medium-content-serif-font,Georgia,Cambria,"Times New Roman",Times,serif';
}

.InlineCommentInputContainer .annotation span{
	background-color: rgba(233, 253, 240, 1) !important;
}

.InlineCommentInputContainer form button{
	margin-right: 10px;
	margin-top: 10px;
}