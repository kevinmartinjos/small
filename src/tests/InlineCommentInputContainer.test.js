import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import InlineCommentInputContainer from '../components/InlineCommentInputContainer';
import ReactTestUtils from 'react-addons-test-utils';

test('InlineCommentInputContainer should be rendered', () => {
	const component = renderer.create(
		<InlineCommentInputContainer content="I am a post" />
	);
	let tree = component.toJSON();
  	expect(tree).toMatchSnapshot();
});
