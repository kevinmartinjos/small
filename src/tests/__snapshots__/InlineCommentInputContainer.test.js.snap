exports[`test InlineCommentInputContainer should be rendered 1`] = `
<div
  className="InlineCommentInputContainer">
  <div />
  <form
    onSubmit={[Function]}>
    <div>
      <textarea
        onChange={[Function]} />
    </div>
    <button
      className="btn btn-sm btn-default"
      disabled={false}
      type="submit">
      Submit
    </button>
    <button
      className="btn btn-sm btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      cancel
    </button>
  </form>
</div>
`;
